urlbd77b745af_KokaKiwi_MCLauncher_tgz-pJ8-net_contrapunctus_lzma_BenchmarkJ8/src/net/contrapunctus/lzma/ConcurrentBufferOutputStream.java

```diff
@@ -0,0 +1,96 @@
+// ConcurrentBufferOutputStream.java -- write bytes to blocking queue
+// Copyright (c)2007 Christopher League <league@contrapunctus.net>
+
+// This is free software, but it comes with ABSOLUTELY NO WARRANTY.
+// GNU Lesser General Public License 2.1 or Common Public License 1.0
+
+package net.contrapunctus.lzma;
+
+import java.io.BufferedOutputStream;
+import java.io.IOException;
+import java.io.InterruptedIOException;
+import java.io.OutputStream;
+import java.io.PrintStream;
+import java.util.concurrent.ArrayBlockingQueue;
+
+class ConcurrentBufferOutputStream extends OutputStream
+{
+    protected ArrayBlockingQueue<byte[]> q;
+    static final int                     BUFSIZE   = 16384;
+    static final int                     QUEUESIZE = 4096;
+    private static final PrintStream     dbg       = System.err;
+    private static final boolean         DEBUG;
+    
+    static
+    {
+        String ds = null;
+        try
+        {
+            ds = System.getProperty("DEBUG_ConcurrentBuffer");
+        }
+        catch (final SecurityException e)
+        {
+        }
+        DEBUG = ds != null;
+    }
+    
+    private ConcurrentBufferOutputStream(ArrayBlockingQueue<byte[]> q)
+    {
+        if (DEBUG)
+        {
+            dbg.printf("%s >> %s%n", this, q);
+        }
+        this.q = q;
+    }
+    
+    static OutputStream create(ArrayBlockingQueue<byte[]> q)
+    {
+        OutputStream out = new ConcurrentBufferOutputStream(q);
+        out = new BufferedOutputStream(out, BUFSIZE);
+        return out;
+    }
+    
+    static ArrayBlockingQueue<byte[]> newQueue()
+    {
+        return new ArrayBlockingQueue<byte[]>(QUEUESIZE);
+    }
+    
+    protected void guarded_put(byte[] a) throws IOException
+    {
+        try
+        {
+            q.put(a);
+        }
+        catch (final InterruptedException exn)
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: InterruptedException is ignored. This can delay thread shutdown and clear the thread’s interrupt status. Only code that implements a thread’s interruption policy can swallow an interruption request.

**Fix**: Rethrow the InterruptedException or reinterrupt the current thread using *Thread.currentThread().interrupt()* so that higher-level interrupt handlers can function correctly.
 If you are wrapping the InterruptedException inside a RuntimeException, call *Thread.currentThread().interrupt()* before throwing the RuntimeException.

Learn more about [interrupts](https://docs.oracle.com/javase/tutorial/essential/concurrency/interrupt.html) and [dealing with InterruptedException](https://www.ibm.com/developerworks/java/library/j-jtp05236/index.html?ca=drs-#N10187)